caiman/source_extraction/cnmf/lazy_arrays.py

```diff
@@ -0,0 +1,444 @@
+
+        # get the mean of each temporal component
+        temporal_mean = np.nanmean(self.temporal, axis=1)
+
+        # get the standard deviation of each temporal component
+        temporal_std = np.nanstd(self.temporal, axis=1)
+
+        # compute mean, max, min, and std projection images
+        self._mean_image = (self.spatial @ temporal_mean).reshape(frame_dims, order="F")
+        self._max_image = (self.spatial @ temporal_max).reshape(frame_dims, order="F")
+        self._min_image = (self.spatial @ temporal_min).reshape(frame_dims, order="F")
+        self._std_image = (self.spatial @ temporal_std).reshape(frame_dims, order="F")
+
+    @property
+    def spatial(self) -> np.ndarray:
+        return self._spatial
+
+    @property
+    def temporal(self) -> np.ndarray:
+        return self._temporal
+
+    @property
+    def n_components(self) -> int:
+        return self._spatial.shape[1]
+
+    @property
+    def n_frames(self) -> int:
+        return self._temporal.shape[1]
+
+    @property
+    def shape(self) -> Tuple[int, int, int]:
+        return self._shape
+
+    @property
+    def min(self) -> float:
+        return self._min
+
+    @property
+    def max(self) -> float:
+        return self._max
+
+    @property
+    def mean_image(self) -> np.ndarray:
+        """mean projection image"""
+        return self._mean_image
+
+    @property
+    def max_image(self) -> np.ndarray:
+        """max projection image"""
+        return self._max_image
+
+    @property
+    def min_image(self) -> np.ndarray:
+        """min projection image"""
+        return self._min_image
+
+    @property
+    def std_image(self) -> np.ndarray:
+        """standard deviation projection image"""
+        return self._std_image
+
+    def _compute_at_indices(self, indices: Union[int, Tuple[int, int]]) -> np.ndarray:
+        rcm = (self.spatial @ self.temporal[:, indices]).reshape(
+            self.shape[1:] + (-1,), order="F"
+        ).transpose([2, 0, 1])
+
+        if rcm.shape[0] == 1:
+            return rcm[0]  # 2d single frame
+        else:
+            return rcm
+
+    def __repr__(self):
+        r = super().__repr__()
+        return f"{r}" \
+               f"n_components: {self.n_components}"
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            raise TypeError(f"cannot compute equality against types that are not {self.__class__.__name__}")
+
+        if (self.spatial == other.spatial) and (self.temporal == other.temporal):
+            return True
+        else:
+            return False
+
+
+# implementation for reconstructed background is identical
+# this is just a subclass to separate them
+# TODO: Is this really necessary? Maybe we just make one class called "Reconstructed Array"
+#  and they keep track of spatial and temporal components that the construct them with?
+class LazyArrayRCB(LazyArrayRCM):
+    """Lazy array for reconstructed background, i.e. b ⊗ f"""
```

Comment:
Similar minor thing with ⊗ here. It's lazily extracting segments of bf  (basically everywhere insert this :smile:)